addons/portal/controllers/mail.py

```diff
@@ -217,9 +91,8 @@ def portal_message_fetch(self, res_model, res_id, domain=False, limit=10, offset
         # Check access
         Message = request.env['mail.message']
         if kw.get('token'):
-            access_as_sudo = _check_special_access(res_model, res_id, token=kw.get('token'))
-            if not access_as_sudo:  # if token is not correct, raise Forbidden
-                raise Forbidden()
+            # Check access to the related thread, if token is not correct this line raises Forbidden
+            request.env[res_model]._get_thread_with_access(res_id, token=kw.get('token'))
```

Comment:
As commented elsewhere, keep this behavior where caller must raise when thread not found.